database.cc

```diff
@@ -1955,7 +1955,12 @@ sstring database::get_available_index_name(const sstring &ks_name, const sstring
     auto base_name = index_metadata::get_default_index_name(cf_name, index_name_root);
     sstring accepted_name = base_name;
     int i = 0;
-    while (existing_names.contains(accepted_name)) {
+    auto name_accepted = [&] {
+        auto index_table_name = secondary_index::index_table_name(accepted_name);
+        const bool table_exists = _ks_cf_to_uuid.find(std::make_pair(ks_name, index_table_name)) != _ks_cf_to_uuid.end();
```

Comment:
We have a function `database::has_schema(std::string_view ks_name, std::string_view cf_name)` which does **exactly** this _ks_cf_to_uuid thing, so you should probably use it for shorter and clearer code. It will be so short you won't even need a variable "table_exists" ;-)

Reply:
I just noticed you use has_schema() in the next patch :-)

Reply:
Right, I found it when coding the second patch and forgot that I used the ugly workaround in the first one. Will fix